### PR TITLE
DDCE-5085: Play migration

### PR DESCRIPTION
**Pull request by robpetterson**
DDCE-5085: Play migration

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -7,15 +7,15 @@ resolvers += Resolver.url("HMRC-open-artefacts-ivy2", url("https://open.artefact
 // Try to remove when sbt 1.8.0+ and scoverage is 2.0.7+
 ThisBuild / libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
 
-addSbtPlugin("uk.gov.hmrc"         % "sbt-auto-build"           % "3.15.0")
-addSbtPlugin("com.typesafe.play"   % "sbt-plugin"               % "2.8.21")
+addSbtPlugin("uk.gov.hmrc"         % "sbt-auto-build"           % "3.18.0")
+addSbtPlugin("org.playframework"   % "sbt-plugin"               % "3.0.0")
```
Comment (mywyau) [inline project/plugins.sbt:11]: this needs to change to 3.0.1 I think or what ever the bot is suggesting

Comment (mywyau) [inline project/plugins.sbt:11]: These two lines this needs to change to whatever the bot is suggesting.

**Comment by mywyau** [comment]
Raised a pr for the build jobs to add .withNodeJs() to passengers frontend jobs. https://github.com/hmrc/build-jobs/pull/16126

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -3,25 +3,24 @@ import sbt.*
...
   private val test           = Seq(
-    "uk.gov.hmrc"         %% "bootstrap-test-play-28"  % bootstrapPlayVersion,
-    "uk.gov.hmrc.mongo"   %% "hmrc-mongo-test-play-28" % hmrcMongoVersion,
-    "org.scalatest"       %% "scalatest"               % "3.2.17",
-    "org.jsoup"            % "jsoup"                   % "1.17.2",
-    "org.mockito"         %% "mockito-scala-scalatest" % "1.17.30",
-    "com.vladsch.flexmark" % "flexmark-all"            % "0.64.8"
+    "uk.gov.hmrc"       %% "bootstrap-test-play-30"  % bootstrapPlayVersion,
+    "uk.gov.hmrc.mongo" %% "hmrc-mongo-test-play-30" % hmrcMongoVersion,
+    "org.scalatest"     %% "scalatest"               % "3.2.17",
```
Comment (lmugnai) [inline project/AppDependencies.scala:21]: scalatest can be removed from here

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -3,25 +3,24 @@ import sbt.*
...
+    "org.jsoup"          % "jsoup"                   % "1.17.2",
```
Comment (lmugnai) [inline project/AppDependencies.scala:22]: jsoup can be removed from here

**Comment by lmugnai** [comment]
@robpetterson please remove .settings(dependencyOverrides += "com.ibm.icu" % "icu4j" % "69.1") from build.sbt

**Comment by lmugnai** [comment]
@robpetterson please remove libraryDependencySchemes ++= Seq("org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always) from build.sbt

File: project/AppDependencies.scala (hunk 2 above)
Comment (robpetterson) [reply project/AppDependencies.scala:21]: removed

File: project/CodeCoverageSettings.scala [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    ".*definition.*",
+    "prod.*",
+    "live.*",
+    "testOnlyDoNotUseInAppConf.*",
+    "app.*",
+    "uk.gov.hmrc.BuildInfo"
+  )
+
+  val settings: Seq[Setting[?]] = Seq(
+    coverageExcludedFiles := "<empty>;.*components.*;.*Routes.*;",
```
Comment (lmugnai) [inline project/CodeCoverageSettings.scala:16]: the excluded packages are being set above. Just use the excludedPackages val here

File: project/CodeCoverageSettings.scala [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+object CodeCoverageSettings {
+  private val excludedPackages: Seq[String] = Seq(
+    "<empty>",
+    ".*definition.*",
+    "prod.*",
+    "live.*",
+    "testOnlyDoNotUseInAppConf.*",
+    "app.*",
+    "uk.gov.hmrc.BuildInfo"
+  )
```
Comment (lmugnai) [inline project/CodeCoverageSettings.scala:13]: can we set these to what they originally were for this repo (<empty>;.*components.*;.*Routes.*;)

Comment (robpetterson) [reply project/CodeCoverageSettings.scala]: Changes made. However, when I run the AC's locally, it's failing on test coverage:

[info] Statement coverage.: 92.41%
[info] Branch coverage....: 85.77%
[info] Coverage reports completed
[error] Coverage is below minimum [92.41% < 99.00%]: Stmt:Total
[error] java.lang.RuntimeException: Coverage minimum was not reached
[error]         at scoverage.CoverageMinimum$All.checkCoverage(CoverageMinimum.scala:76)
[error]         at scoverage.ScoverageSbtPlugin$.$anonfun$coverageReport0$1(ScoverageSbtPlugin.scala:240)
[error]         at scoverage.ScoverageSbtPlugin$.$anonfun$coverageReport0$1$adapted(ScoverageSbtPlugin.scala:211)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error]         at sbt.Execute.work(Execute.scala:292)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error]         at java.base/java.lang.Thread.run(Thread.java:829)
[error] (coverageReport) Coverage minimum was not reached


How should I handle this?

**Review by lmugnai — APPROVED** [review]

**Pull request merged**